FantasyUniverse/Networking/Cache.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Cache.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 19.04.2022.
+//
+
+import Foundation
+
+protocol Cachable {
+    associatedtype Element
+    associatedtype Searched
+        
+    func getElements() -> [Element]
+    func cacheElements(_ objects: [Element])
```

Comment:
This can be implemented with `cleanCache` and `addToCache` in a protocol extension so that every type implementing this protocol gets it for free